app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -208,9 +208,12 @@ def current_issuer
   end
 
   def request_url
-    url = URI.parse request.original_url
-    url.path = remap_auth_post_path(url.path)
-    query_params = Rack::Utils.parse_nested_query url.query
+    url = URI.parse(request.original_url)
+
+    # we need to grab just the .path, in case the full value includes query params like ?locale=
+    url.path = URI(api_saml_auth_path(path_year: params[:path_year])).path
+
+    query_params = Rack::Utils.parse_nested_query(url.query)
```

Comment:
Wondering if there might be a way to avoid the double parse here. Could we generate the full URL instead? Plus use `request.query_parameters` directly?

```suggestion
    url = URI(api_saml_auth_url(path_year: params[:path_year]))
    
    query_params = request.query_parameters
```

Reply:
will give it a shot

Reply:
ok so by running the feature specs, I think I know why it was done this way, to preserve the host, ran into a few of these. Going to see if I can copy the host over directly, and if not, will revert this patch

```
      ActionController::Redirecting::UnsafeRedirectError:
        Unsafe redirect to "http://www.example.com/", pass allow_other_host: true to redirect anyway.
```

Reply:
ok so that spec failure was a red herring, kept this change in (fixed confounding bug in 2a844b4d30e42a49e0edccc4dbf47cdc270edcdd)